homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -11,7 +11,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-GATEWAY_PLATFORMS = ["alarm_control_panel"]
+GATEWAY_PLATFORMS = ["alarm_control_panel", "sensor"]
```

Comment:
In line 27 of the __init__.py file it is checked what kind of CONF_FLOW_TYPE it is, if it is not the gateway but the vacuum, it will split of there and vacuum can define its own list of VACUUM_PLATFORMS.

in line 63 of sensor.py file inside the async_setup_entry function it is again checked what kind of CONF_FLOW_TYPE it is, if it is not the gateway but the vacuum it will split of again, adding the entities based on the (still to be made) VacuumSensor class instead of the XiaomiGatewaySensor class